db/crud.go

```diff
@@ -476,39 +476,63 @@ func (db *DatabaseCollectionWithUser) GetDelta(ctx context.Context, docID, fromR
 	// If the fromRevision is a removal cache entry (no body), but the user has access to that removal, then just
 	// return 404 missing to indicate that the body of the revision is no longer available.
 	// Delta can't be generated if we don't have the fromRevision body.
-	if fromRevision.Removed {
+	if initialFromRevision.Removed {
 		return nil, nil, ErrMissing
 	}
 
 	// If the fromRevision was a tombstone, then return error to tell delta sync to send full body replication
-	if fromRevision.Deleted {
+	if initialFromRevision.Deleted {
 		return nil, nil, base.ErrDeltaSourceIsTombstone
 	}
 
-	// If both body and delta are not available for fromRevId, the delta can't be generated
-	if fromRevision.BodyBytes == nil && fromRevision.Delta == nil {
-		return nil, nil, err
+	// locking ensures clients requesting deltas sharing the same from revision memoize the work to generate the delta and share the cached result
+	initialFromRevision.RevCacheValueDeltaLock.RLock()
```

Comment:
We already have separate synchronization for actually updating the delta in the rev cache (in UpdateDelta) - I believe the intention is that DocumentRevision.Delta points to an immutable delta, and in UpdateDelta a new delta is stored and referenced.  Given that, do we actually need to acquire a read lock on RevCacheValueDeltaLock, if the only intention is to prevent concurrent delta computation?  Is it sufficient to just acquire the write lock before starting the work to fetch/compute the delta?